### ISSUE TITLE
[Bug]: playwright merge-reports ignores output

### ISSUE THREAD

**Issue by huehnerlady** [microsoft/playwright#33237]
### Version

1.48.1

### Steps to reproduce

1. run e2e tests with shards using the config:  
<details>
  <summary>playwright config</summary>

```ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  // Look for test files in the "tests" directory, relative to this configuration file.
  testDir: 'e2e-tests',

  // Exclude specific test files or directories.
  testIgnore: ['e2e-tests/test-*.spec.ts'],

  // Run all tests in parallel.
  fullyParallel: true,

  // Fail the build on CI if you accidentally left test.only in the source code.
  forbidOnly: !!process.env['CI'],

  // Retry on CI only.
  retries: process.env['CI'] ? 2 : 0,

  // Opt out of parallel tests on CI.
  workers: process.env['CI'] ? 1 : undefined,

  // Reporter to use
  reporter: [
    ['list'], // You can combine multiple reporters
    ['blob'],
    ['html', { outputFile: './playwright-report/report.html' }],
    ['json', { outputFile: './playwright-report/report.json' }],
    ['junit', { outputFile: './playwright-report/report.xml' }],
    ['playwright-ctrf-json-reporter', {}],
  ],

  use: {
    // Base URL to use in actions like `await page.goto('/')`.
    baseURL: 'http://localhost:4200/',

    // Collect trace when retrying the failed test.
    trace: 'on-first-retry',
  },
  // Configure projects for major browsers.
  projects: [
    {
      name: 'Desktop Chrome',
      use: { ...devices['Desktop Chrome'] },
    },
    {
      name: 'Mobile Safari',
      use: {
        ...devices['iPhone 12 Mini'],
      },
    },
  ],
  // Run your local dev server before starting the tests.
  webServer: {
    command: 'npm run start',
    url: 'http://localhost:4200/',
    reuseExistingServer: !process.env['CI'],
    stdout: 'pipe',
    timeout: 120 * 1000,
  },
});
```
</details>

<details>
  <summary>Github action</summary>

```yaml
name: Playwright Tests
on: [ pull_request ]

permissions:
  # creates a new check with unit test results
  checks: write
  # needs to git diff the repo
  contents: read
  # find the pr for this commit
  issues: read
  # write artifacts and comment on pr related to this
  pull-requests: write

jobs:
  e2e-tests:
    name: E2E Tests
    concurrency:
      group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}-${{ matrix.shardIndex }}
      cancel-in-progress: true
    timeout-minutes: 60
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        shardIndex: [1, 2]
        shardTotal: [2]
    steps:
    - uses: actions/checkout@v4
    - uses: actions/setup-node@v4
      with:
        node-version: lts/*
    - name: Install dependencies
      run: npm ci --include=optional
    - name: Setup Playwright
      run: npx playwright install --with-deps
    - name: Run Playwright Tests
      run: npx playwright test  --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
    - name: Upload blob report to GitHub Actions Artifacts
      if: ${{ !cancelled() }}
      uses: actions/upload-artifact@v4
      with:
        name: blob-report-${{ matrix.shardIndex }}.zip
        path: blob-report/report-${{ matrix.shardIndex }}.zip
        retention-days: 1
  merge-reports:
    # Merge reports after playwright-tests, even if some shards have failed
    if: ${{ !cancelled() }}
    needs: [e2e-tests]

    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - uses: actions/setup-node@v4
      with:
        node-version: lts/*
    - name: Install dependencies
      run: npm ci

    - name: Download blob reports from GitHub Actions Artifacts
      uses: actions/download-artifact@v4
      with:
        path: all-blob-reports
        pattern: blob-report-*
        merge-multiple: true
    - name: Merge into HTML Report
      run: npx playwright merge-reports --reporter html ./all-blob-reports
    - name: Merge into xml Report
      run: npx playwright merge-reports --reporter junit ./all-blob-reports
    - name: Merge into xml Report (version 2)
      run: npx playwright merge-reports --reporter junit ./all-blob-reports >> playwright-report/report.xml
    - name: Merge ctrf reports
      run: npx playwright merge-reports --reporter playwright-ctrf-json-reporter ./all-blob-reports
    - name: Add ctrf report to summary
      run: npx github-actions-ctrf all ctrf/ctrf-report.json
    - name: Upload HTML report
      uses: actions/upload-artifact@v4
      with:
        name: report--attempt-${{ github.run_attempt }}
        path: playwright-report
        retention-days: 14
    - name: Test Report
      uses: dorny/test-reporter@v1.9.1
      with:
        name: E2E Test Report
        path: playwright-report/report.xml
        reporter: jest-junit
```
</details>

2. run `npx playwright merge-reports --reporter junit ` on the folder where all blob-reports are

### Expected behavior

When I run `npx playwright merge-reports --reporter junit` I expect it to use the output file put down in the config

### Actual behavior

The result of the merge is printed into stdout

### Additional context

for the `html` reporter it seems to work, for the `playwright-ctrf-json-reporter` the file is also written. Only for junit and json the result is printed to stdout. My work around is that I do not use multiple reporters in one command, but split it up and pipe it into a text file:

`npx playwright merge-reports --reporter junit >> playwright-report/report.xml`

### Environment

Github action:
<details>
  <summary>Details</summary>

name: Playwright Tests
on: [ pull_request ]

permissions:
  # creates a new check with unit test results
  checks: write
  # needs to git diff the repo
  contents: read
  # find the pr for this commit
  issues: read
  # write artifacts and comment on pr related to this
  pull-requests: write

jobs:
  e2e-tests:
    name: E2E Tests
    concurrency:
      group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}-${{ matrix.shardIndex }}
      cancel-in-progress: true
    timeout-minutes: 60
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        shardIndex: [1, 2]
        shardTotal: [2]
    steps:
    - uses: actions/checkout@v4
    - uses: actions/setup-node@v4
      with:
        node-version: lts/*
    - name: Install dependencies
      run: npm ci --include=optional
    - name: Setup Playwright
      run: npx playwright install --with-deps
    - name: Run Playwright Tests
      run: npx playwright test  --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
    - name: Upload blob report to GitHub Actions Artifacts
      if: ${{ !cancelled() }}
      uses: actions/upload-artifact@v4
      with:
        name: blob-report-${{ matrix.shardIndex }}.zip
        path: blob-report/report-${{ matrix.shardIndex }}.zip
        retention-days: 1
  merge-reports:
    # Merge reports after playwright-tests, even if some shards have failed
    if: ${{ !cancelled() }}
    needs: [e2e-tests]

    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - uses: actions/setup-node@v4
      with:
        node-version: lts/*
    - name: Install dependencies
      run: npm ci

    - name: Download blob reports from GitHub Actions Artifacts
      uses: actions/download-artifact@v4
      with:
        path: all-blob-reports
        pattern: blob-report-*
        merge-multiple: true
    - name: Merge into HTML Report
      run: npx playwright merge-reports --reporter html ./all-blob-reports
    - name: Merge into xml Report
      run: npx playwright merge-reports --reporter junit ./all-blob-reports
    - name: Merge into xml Report (version 2)
      run: npx playwright merge-reports --reporter junit ./all-blob-reports >> playwright-report/report.xml
    - name: Merge ctrf reports
      run: npx playwright merge-reports --reporter playwright-ctrf-json-reporter ./all-blob-reports
    - name: Add ctrf report to summary
      run: npx github-actions-ctrf all ctrf/ctrf-report.json
    - name: Upload HTML report
      uses: actions/upload-artifact@v4
      with:
        name: report--attempt-${{ github.run_attempt }}
        path: playwright-report
        retention-days: 14
    - name: Test Report
      uses: dorny/test-reporter@v1.9.1
      with:
        name: E2E Test Report
        path: playwright-report/report.xml
        reporter: jest-junit

</details>

**Comment by mxschmitt** [comment]
By default `npx playwright merge-reports` won't consume a config, if you want to use a config, you can do

    `npx playwright merge-reports --config .github/workflows/merge.config.ts`

see [here](https://playwright.dev/docs/test-sharding#merge-reports-cli). We recommend creating a separate merge config.

**Comment by huehnerlady** [comment]
@mxschmitt thanks for the answer. I did try to hand in that config as mentioned in your link, but it did not change anything.

Why do you recommend having a separate config?

**Comment by mxschmitt** [comment]
You usually want to have separate reports configured for after the merge vs. when you run the actual tests. In your case you could change:

```
  reporter: [
    ['list'], // You can combine multiple reporters
    ['blob'],
    ['html', { outputFile: './playwright-report/report.html' }],
    ['json', { outputFile: './playwright-report/report.json' }],
    ['junit', { outputFile: './playwright-report/report.xml' }],
    ['playwright-ctrf-json-reporter', {}],
  ],
```

to just `reporter: 'blob'` and move this whole section above `^` into the `merge.config.ts`.

**Comment by mxschmitt** [comment]
You can remove the `--reporter=junit` part since you already specify it in the config, then it should work.

**Comment by huehnerlady** [comment]
Ah I see thanks. This did work. I did not know that the one parameter overrides the other 👍🏻

**Comment by mxschmitt** [comment]
Awesome, will close by that.
